### PR TITLE
feat(notes): neutralise GitHub auto-linked refs and mentions in changelogs

### PR DESCRIPTION
**Pull request by goosewobbler**
## What

Changelog output (the standing-PR body, `CHANGELOG.md`, and the GitHub release body) contains bare tokens GitHub auto-links:

- **Issue/PR refs** (`#481`) — auto-linked with hovercard previews, clutter in expanded changelogs.
- **Scoped package names / mentions** (`@wdio/native-cdp-bridge`, `@user`) — parsed as `@org/team` mentions → stray links that can **ping or subscribe a real org/team on every release PR**.

## Changes

- New `notes.changelog.refs` config — `strip` | `escape` | `link` (default `link`):
  - `link`: `[#NNN](https://github.com/owner/repo/issues/NNN)` — clickable, keeps the hovercard, no longer a bare token. Always points at `/issues/NNN` (GitHub redirects issues↔pulls). Falls back to `escape` for a missing/non-GitHub repo URL.
  - `escape`: literal `\#NNN` (no link, no hovercard).
  - `strip`: refs removed entirely.
- `@scope/pkg` / `@user` mentions in entry descriptions are **always** backslash-escaped, regardless of `refs`. Conservative: emails (`foo@bar.com`), mid-word `@`, and mentions inside inline code spans are left untouched.
- Two shared pure helpers (`renderIssueRefs`, `escapeChangelogMentions`) in `@releasekit/core` drive both the notes Markdown and standing-PR surfaces, so the treatment is identical. The GitHub release body (built via `renderMarkdown`) inherits link-mode refs + mention escaping for free.
- Config schema + generated docs regenerated.

## Out of scope (follow-up)
A **third** render surface — `packages/release/src/preview/format.ts` (the feeder-PR / preview comment) — has the same bare-`#NNN` + un-escaped-`@` problem but isn't in this issue's named scope and needs `repoUrl`/`refs` threaded through the preview pipeline. Left untouched; worth a follow-up issue.

## Tests

Core helper (all three modes incl. unparseable-repoUrl fallback; mention escaping incl. email and code-span cases), notes Markdown rendering, and standing-PR changelog-region rendering. Full gate green (32/32); `schema:check` clean.

Closes #499

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<details open><summary><h3>Greptile Summary</h3></summary>

This PR neutralises the two GitHub auto-linking hazards in changelog output — bare `#NNN` refs and `@scope/user` mentions — across all three render surfaces (root/per-package `CHANGELOG.md`, standing-PR body, and GitHub release body). A new `notes.changelog.refs` config (`link` | `escape` | `strip`, default `link`) controls how issue refs are rendered; `@`-mention escaping is unconditional.

- Two pure helpers (`renderIssueRefs`, `escapeChangelogMentions`) are added to `@releasekit/core` and plumbed through `markdown.ts`, `aggregator.ts`, `changelog-region.ts`, and `standing-pr.ts`; the previous P1 gaps (unescaped scope/leadIn/scope-header fields, `refs` not forwarded to per-package monorepo changelogs) addressed in earlier review rounds are correctly closed.
- The `escapeChangelogMentions` regex uses a single-backtick code-span guard to avoid polynomial backtracking on adversarial backtick runs (ReDoS) — this is verified by a dedicated test.
- The `context.enhanced?.summary` LLM prose field is not passed through `escapeChangelogMentions`; the PR description explicitly scopes to "entry descriptions", so this is a known gap rather than a regression.
</details>

<h3>Confidence Score: 5/5</h3>

Safe to merge — the `refs` config is correctly threaded through all named render surfaces, mention escaping covers scope/leadIn/scope-header fields, and the ReDoS concern is explicitly guarded and tested.

The core helpers are pure functions with thorough unit tests including edge cases (emails, mid-word @, already-escaped tokens, code spans, adversarial backtick runs). The `refs` setting is correctly forwarded through the changelog file writer, the monorepo aggregator, and both standing-PR render functions. The only gap — `enhanced.summary` not being escaped — is explicitly scoped out in the PR description and is pre-existing LLM-generated prose, not changelog entries.

No files require special attention. The one P2 observation about `context.enhanced?.summary` in `markdown.ts` is a known coverage gap documented in the PR description, not a regression introduced here.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/core/src/changelogRefs.ts | New module housing two pure helpers: `renderIssueRefs` (three-mode ref rendering with GitHub URL parsing) and `escapeChangelogMentions` (lookbehind regex guards emails, mid-word @, already-escaped, and backtick spans). ReDoS concern is explicitly addressed via single-backtick code-span pattern. |
| packages/notes/src/output/markdown.ts | Adds `refs`/`repoUrl` threading through `EntryRefOptions`, applies `escapeChangelogMentions` to `entry.description`, `entry.scope`, and `entry.leadIn` in `formatEntry`, and escapes the grouped-scope header. Both LLM-categorised and non-LLM entry paths now receive `refOpts`. |
| packages/notes/src/core/pipeline.ts | Replaces local `parseOwnerRepo` with the shared `parseGitHubOwnerRepo`, builds `fmtOpts.refs` from `changelogConfig.refs ?? 'link'`, and forwards `refs` down to `writeMonorepoFiles`/`writeMonorepoChangelogs` so per-package changelogs honour the setting. |
| packages/release/src/standing-pr/changelog-region.ts | Introduces `RefRenderOptions`, escapes `entry.description` via `escapeChangelogMentions`, replaces bare `issueIds.join` with `renderIssueRefs`, and propagates `refOpts` through `entryLine`/`renderGrouped`/`renderCombinedFooter`/`makeRowChangelogRenderer`. The existing backtick-wrapped scope field is left as-is (safe by design). |
| packages/release/src/standing-pr/standing-pr.ts | Reads `changelogConfig.refs` (with truthy-check fallback to `'link'` for `false`/absent configs) and passes `changelogRefsMode` into both `makeRowChangelogRenderer` and both `renderCombinedFooter` call sites. |
| packages/notes/src/monorepo/aggregator.ts | Accepts `refs: ChangelogRefsMode = 'link'` parameter, threads it into `fmtOpts` for the root changelog and into `prependVersion`/`renderMarkdown` calls for per-package changelogs. |

</details>

<details><summary><h3>Flowchart</h3></summary>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A["changelog.refs config\n('link' | 'escape' | 'strip')"] --> B["pipeline.ts: fmtOpts.refs"]

    B --> C["writeMarkdown (root CHANGELOG.md)"]
    B --> D["writeMonorepoFiles → aggregator.ts"]
    D --> E["renderMarkdown / prependVersion\n(per-package CHANGELOG.md)"]

    B --> F["formatVersion (release notes file)"]

    G["standing-pr.ts: changelogRefsMode"] --> H["makeRowChangelogRenderer\n(per-row details pane)"]
    G --> I["renderCombinedFooter\n(combined footer)"]

    J["renderIssueRefs(issueIds, mode, repoUrl)"]
    K["escapeChangelogMentions(text)"]

    C --> J
    E --> J
    F --> J
    H --> J
    I --> J

    C --> K
    E --> K
    F --> K
    H --> K
    I --> K

    L["repoUrl"] --> J
    M["parseGitHubOwnerRepo(url)"] --> J
    L --> M
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
flowchart TD
    A["changelog.refs config\n('link' | 'escape' | 'strip')"] --> B["pipeline.ts: fmtOpts.refs"]

    B --> C["writeMarkdown (root CHANGELOG.md)"]
    B --> D["writeMonorepoFiles → aggregator.ts"]
    D --> E["renderMarkdown / prependVersion\n(per-package CHANGELOG.md)"]

    B --> F["formatVersion (release notes file)"]

    G["standing-pr.ts: changelogRefsMode"] --> H["makeRowChangelogRenderer\n(per-row details pane)"]
    G --> I["renderCombinedFooter\n(combined footer)"]

    J["renderIssueRefs(issueIds, mode, repoUrl)"]
    K["escapeChangelogMentions(text)"]

    C --> J
    E --> J
    F --> J
    H --> J
    I --> J

    C --> K
    E --> K
    F --> K
    H --> K
    I --> K

    L["repoUrl"] --> J
    M["parseGitHubOwnerRepo(url)"] --> J
    L --> M
```

</a>
</details>

<details><summary><h3>Comments Outside Diff (2)</h3></summary>

1. `packages/notes/src/output/markdown.ts`, line 120-125 ([link](https://github.com/goosewobbler/releasekit/blob/7e1c4945b72cd27ca6992cfd94e92ac05fb1dc4a/packages/notes/src/output/markdown.ts#L120-L125)) 

   <a href="#">P1</a> **Scope group header also emits unescaped `@` mentions**

   When multiple entries share the same scoped-package scope, a group header line `**${entry.scope}**:` is emitted here. If the scope is `@wdio/native-cdp-bridge`, the resulting `**@wdio/native-cdp-bridge**:` header will trigger the same GitHub org-mention as a bare `@wdio` in prose. `escapeChangelogMentions(entry.scope)` needs to be applied before this interpolation as well.

   <a href="https://app.greptile.com/ide/claude-code?prompt=This%20is%20a%20comment%20left%20during%20a%20code%20review.%0APath%3A%20packages%2Fnotes%2Fsrc%2Foutput%2Fmarkdown.ts%0ALine%3A%20120-125%0A%0AComment%3A%0A**Scope%20group%20header%20also%20emits%20unescaped%20%60%40%60%20mentions**%0A%0AWhen%20multiple%20entries%20share%20the%20same%20scoped-package%20scope%2C%20a%20group%20header%20line%20%60**%24%7Bentry.scope%7D**%3A%60%20is%20emitted%20here.%20If%20the%20scope%20is%20%60%40wdio%2Fnative-cdp-bridge%60%2C%20the%20resulting%20%60**%40wdio%2Fnative-cdp-bridge**%3A%60%20header%20will%20trigger%20the%20same%20GitHub%20org-mention%20as%20a%20bare%20%60%40wdio%60%20in%20prose.%20%60escapeChangelogMentions%28entry.scope%29%60%20needs%20to%20be%20applied%20before%20this%20interpolation%20as%20well.%0A%0AHow%20can%20I%20resolve%20this%3F%20If%20you%20propose%20a%20fix%2C%20please%20make%20it%20concise.&repo=goosewobbler%2Freleasekit&pr=503&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInClaudeDark.svg?v=3"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInClaude.svg?v=3">Fix in Claude Code</picture></a> <a href="https://app.greptile.com/api/ide/cursor?prompt=This%20is%20a%20comment%20left%20during%20a%20code%20review.%0APath%3A%20packages%2Fnotes%2Fsrc%2Foutput%2Fmarkdown.ts%0ALine%3A%20120-125%0A%0AComment%3A%0A**Scope%20group%20header%20also%20emits%20unescaped%20%60%40%60%20mentions**%0A%0AWhen%20multiple%20entries%20share%20the%20same%20scoped-package%20scope%2C%20a%20group%20header%20line%20%60**%24%7Bentry.scope%7D**%3A%60%20is%20emitted%20here.%20If%20the%20scope%20is%20%60%40wdio%2Fnative-cdp-bridge%60%2C%20the%20resulting%20%60**%40wdio%2Fnative-cdp-bridge**%3A%60%20header%20will%20trigger%20the%20same%20GitHub%20org-mention%20as%20a%20bare%20%60%40wdio%60%20in%20prose.%20%60escapeChangelogMentions%28entry.scope%29%60%20needs%20to%20be%20applied%20before%20this%20interpolation%20as%20well.%0A%0AHow%20can%20I%20resolve%20this%3F%20If%20you%20propose%20a%20fix%2C%20please%20make%20it%20concise.&pr=503&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInCursorDark.svg?v=3"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInCursor.svg?v=3">Fix in Cursor</picture></a>

2. `packages/notes/src/core/pipeline.ts`, line 479-481 ([link](https://github.com/goosewobbler/releasekit/blob/c9c1cf96f7b6634d286c4440656b1dddf3c2a341/packages/notes/src/core/pipeline.ts#L479-L481)) 

   <a href="#">P1</a> **`refs` config silently ignored for per-package monorepo changelogs**

   `writeMonorepoFiles` does not receive `fmtOpts` (which carries `refs`), so when `changelog.mode` is `'packages'` or `'both'`, every per-package `CHANGELOG.md` is always rendered with `link` mode regardless of the user's configured `refs` value. `writeMonorepoChangelogs` in `aggregator.ts` also hard-codes its own `{ includePackageName: true }` options and calls both `prependVersion` and `renderMarkdown` without forwarding any format options. A user who sets `notes.changelog.refs: 'strip'` would see the root changelog correctly strip refs (since `writeMarkdown` at line 474 does receive `fmtOpts`) but every per-package file would still emit full Markdown issue links.

   <a href="https://app.greptile.com/ide/claude-code?prompt=This%20is%20a%20comment%20left%20during%20a%20code%20review.%0APath%3A%20packages%2Fnotes%2Fsrc%2Fcore%2Fpipeline.ts%0ALine%3A%20479-481%0A%0AComment%3A%0A**%60refs%60%20config%20silently%20ignored%20for%20per-package%20monorepo%20changelogs**%0A%0A%60writeMonorepoFiles%60%20does%20not%20receive%20%60fmtOpts%60%20%28which%20carries%20%60refs%60%29%2C%20so%20when%20%60changelog.mode%60%20is%20%60'packages'%60%20or%20%60'both'%60%2C%20every%20per-package%20%60CHANGELOG.md%60%20is%20always%20rendered%20with%20%60link%60%20mode%20regardless%20of%20the%20user's%20configured%20%60refs%60%20value.%20%60writeMonorepoChangelogs%60%20in%20%60aggregator.ts%60%20also%20hard-codes%20its%20own%20%60%7B%20includePackageName%3A%20true%20%7D%60%20options%20and%20calls%20both%20%60prependVersion%60%20and%20%60renderMarkdown%60%20without%20forwarding%20any%20format%20options.%20A%20user%20who%20sets%20%60notes.changelog.refs%3A%20'strip'%60%20would%20see%20the%20root%20changelog%20correctly%20strip%20refs%20%28since%20%60writeMarkdown%60%20at%20line%20474%20does%20receive%20%60fmtOpts%60%29%20but%20every%20per-package%20file%20would%20still%20emit%20full%20Markdown%20issue%20links.%0A%0AHow%20can%20I%20resolve%20this%3F%20If%20you%20propose%20a%20fix%2C%20please%20make%20it%20concise.&repo=goosewobbler%2Freleasekit&pr=503&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInClaudeDark.svg?v=3"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInClaude.svg?v=3">Fix in Claude Code</picture></a> <a href="https://app.greptile.com/api/ide/cursor?prompt=This%20is%20a%20comment%20left%20during%20a%20code%20review.%0APath%3A%20packages%2Fnotes%2Fsrc%2Fcore%2Fpipeline.ts%0ALine%3A%20479-481%0A%0AComment%3A%0A**%60refs%60%20config%20silently%20ignored%20for%20per-package%20monorepo%20changelogs**%0A%0A%60writeMonorepoFiles%60%20does%20not%20receive%20%60fmtOpts%60%20%28which%20carries%20%60refs%60%29%2C%20so%20when%20%60changelog.mode%60%20is%20%60'packages'%60%20or%20%60'both'%60%2C%20every%20per-package%20%60CHANGELOG.md%60%20is%20always%20rendered%20with%20%60link%60%20mode%20regardless%20of%20the%20user's%20configured%20%60refs%60%20value.%20%60writeMonorepoChangelogs%60%20in%20%60aggregator.ts%60%20also%20hard-codes%20its%20own%20%60%7B%20includePackageName%3A%20true%20%7D%60%20options%20and%20calls%20both%20%60prependVersion%60%20and%20%60renderMarkdown%60%20without%20forwarding%20any%20format%20options.%20A%20user%20who%20sets%20%60notes.changelog.refs%3A%20'strip'%60%20would%20see%20the%20root%20changelog%20correctly%20strip%20refs%20%28since%20%60writeMarkdown%60%20at%20line%20474%20does%20receive%20%60fmtOpts%60%29%20but%20every%20per-package%20file%20would%20still%20emit%20full%20Markdown%20issue%20links.%0A%0AHow%20can%20I%20resolve%20this%3F%20If%20you%20propose%20a%20fix%2C%20please%20make%20it%20concise.&pr=503&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInCursorDark.svg?v=3"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInCursor.svg?v=3">Fix in Cursor</picture></a>
</details>

<sub>Reviews (3): Last reviewed commit: ["fix(notes): forward refs mode to per-pac..."](https://github.com/goosewobbler/releasekit/commit/481fcae16dc446177d31dd683bcb3ec0aaf3e369) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=40793020)</sub>

### CODE REVIEW

File: packages/notes/src/output/markdown.ts [hunk 1]
```diff
@@ -49,31 +49,43 @@ function groupEntriesByType(entries: ChangelogEntry[]): Map<ChangelogEntry['type
...
   } else if (entry.breaking) {
-    line = `- **BREAKING** ${entry.description}`;
+    line = `- **BREAKING** ${description}`;
   } else if (entry.scope && !opts?.hideScope) {
-    line = `- **${entry.scope}**: ${entry.description}`;
+    line = `- **${entry.scope}**: ${description}`;
   } else {
-    line = `- ${entry.description}`;
+    line = `- ${description}`;
   }
```
[the remark this answers is not in the document]
Comment (goosewobbler) [reply packages/notes/src/output/markdown.ts]: Fixed in c9c1cf96. `escapeChangelogMentions` now also wraps `entry.scope` and `entry.leadIn` in `formatEntry`, plus the grouped scope header. A scoped package like `@wdio/native-cdp-bridge` used as a scope no longer mentions `@wdio`. (The standing-PR surface backticks the scope, so it was already safe.) Added tests for scope, leadIn, and the group-header path.
